src/framework/ui/tabNavigator/bottomTabNavigator.component.tsx

```diff
@@ -0,0 +1,147 @@
+import React from 'react';
+import {
+  Dimensions,
+  View,
+  Text,
+  ViewProps,
+  StyleSheet,
+  TouchableWithoutFeedback,
+  TouchableWithoutFeedbackProps,
+} from 'react-native';
+import {
+  StyledComponentProps,
+  StyleType,
+} from '@kitten/theme';
+import { TabBarIndicator } from '../tab/tabBarIndicator.component';
+
+export interface TabRoute {
+  routeName: string;
+  routeImage?: React.ReactElement<any>;
+}
+
+export const TAB_TEST_ID = '@tab-navigator/tab-';
+
+interface TabNavigatorProps {
+  appearance?: string;
+  routes: TabRoute[];
+  currentIndex?: number;
+  onTabChoose?: (routeName: string) => void;
+}
+
+export type Props = TabNavigatorProps & StyledComponentProps & ViewProps;
+
+export class BottomTabNavigator extends React.Component<Props> {
+
+  static defaultProps: Partial<Props> = {
+    currentIndex: 0,
+  };
+
+  private getComponentStyle = (style: StyleType): StyleType => {
+    return {
```

Comment:
why do we use appearances as style holders for specific parts of component?

Reply:
I think this is generally incorrect way of using mappings.
Due to complex token groups are not implemented yet, consider implementing composite property names instead of declaring them in separate objects: `tabContent.color -> tabContentColor`

Reply:
Got it. Thanks!